app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php

```diff
@@ -0,0 +1,232 @@
+
+    /**
+     * @var PaymentTokenInterface|MockObject
+     */
+    private $paymentTokenMock;
+
+
+    /**
+     * @var ResolveInfo|MockObject
+     */
+    private $resolveInfoMock;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp() : void
+    {
+        $this->objectManager = new ObjectManager($this);
+
+        $this->contextMock = $this->getMockBuilder(ContextInterface::class)
+            ->disableOriginalConstructor()
+            ->setMethods(
+                [
+                    'getExtensionAttributes',
+                    'getUserId',
+                    'getUserType',
+                ]
+            )
+            ->getMock();
+
+        $this->contextExtensionMock = $this->getMockBuilder(ContextExtensionInterface::class)
+            ->setMethods(
+                [
+                    'getIsCustomer',
+                    'getStore',
+                    'setStore',
+                    'setIsCustomer',
+                ]
+            )
+            ->getMock();
+
+        $this->fieldMock = $this->getMockBuilder(Field::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->paymentTokenManagementMock = $this->getMockBuilder(PaymentTokenManagementInterface::class)
+            ->getMockForAbstractClass();
+
+        $this->paymentTokenRepositoryMock = $this->getMockBuilder(PaymentTokenRepositoryInterface::class)
+            ->getMockForAbstractClass();
+
+        $this->paymentTokenMock = $this->getMockBuilder(PaymentTokenInterface::class)
+            ->getMockForAbstractClass();
+
+        $this->resolveInfoMock = $this->getMockBuilder(ResolveInfo::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->resolver = $this->objectManager->getObject(
+            DeletePaymentToken::class,
+            [
+                'paymentTokenManagement' => $this->paymentTokenManagementMock,
+                'paymentTokenRepository' => $this->paymentTokenRepositoryMock,
+            ]
+        );
+    }
+
+    /**
+     * Test delete customer payment token
+     */
+    public function testDeleteCustomerPaymentToken()
+    {
+        $isCustomer = true;
+        $paymentTokenResult = true;
+
+        $this->contextMock
+            ->expects($this->once())
+            ->method('getExtensionAttributes')
+            ->willReturn($this->contextExtensionMock);
+
+        $this->contextExtensionMock
+            ->expects($this->once())
+            ->method('getIsCustomer')
+            ->willReturn($isCustomer);
+
+        $this->paymentTokenManagementMock
+            ->expects($this->once())
+            ->method('getByPublicHash')
+            ->willReturn($this->paymentTokenMock);
+
+        $this->paymentTokenRepositoryMock
+            ->expects($this->once())
+            ->method('delete')
+            ->with($this->paymentTokenMock)
+            ->willReturn($paymentTokenResult);
+
+        $this->resolver->resolve(
```

Comment:
I would recommend using assertions (`self::assert`) when a testable method is supposed to return something.

Reply:
Good notice!

**implemented**

Thanks @rogyar